### PR TITLE
Root group IDs, shareKeys and share link compatibility ADR

### PR DESCRIPTION
**Pull request by nf-s**
### Sharekeys ADR draft

Related to https://github.com/TerriaJS/terriajs/issues/4774 https://github.com/TerriaJS/terriajs/pull/5055 https://github.com/TerriaJS/terriajs/pull/5042

### CODE REVIEW

**Comment by nf-s** [comment]
Nick TODO:

- [x] Create scenarios of catalog changes over time: for example v7 TerriaMap -> v8 TerriaMap (through catalog-converter) -> v8 Terrace/Magda JSON -> v8 Magda catalog
- [x] Show id changes over time
- [x] What shareKeys do we need to add at each step?
- [x] How does this impact general users?

File: architecture/0005-root-group-ids-and-shareKeys.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
+# 5. Root group IDs, `shareKeys` and share link compatibility
+
+Date: 2020-12-03  
+Version: 2
+
+## Status
+
+In discussion
+
+## Context
```
Comment (KeyboardSounds) [inline architecture/0005-root-group-ids-and-shareKeys.md:10]: Add:
```
### Glossary
**v7 JSON catalog**: An old v7 catalog that lives in a JSON file, typically specified in a TerriaMap's config.json.
**v8 JSON catalog**: Like an old v7 catalog, but v8.
**v8 JSON catalog loaded from Magda**: A v8 json catalog sitting in the map-config record for that map in Magda.
**Magda catalog**: Each catalog item and group is a magda record.
**Root group**: The group of a catalog that contains all other groups and catalog items in a catalog.
```

**Comment by KeyboardSounds** [comment]
I think we need to make a great big table comparing the pros and cons of each of our options, otherwise we're never going to make a decision. like

### Decision: what to eat for dinner
| option | pros | cons |
|-------|------|------|
| Maccas | delicious | funding an evil megacorporation |

**Comment by nf-s** [comment]
Defamation lawsuit incoming

File: architecture/0005-root-group-ids-and-shareKeys.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+| option | pros | cons |
+|-------|------|------|
+| Write v7 autoID `shareKeys` | Can change catalog structure and preserve old v7 share links.<br/>Can remove opaque autoID conversion in share link conversion | Makes catalog JSON less readable, may be confusing to some users. | 
+| Create random IDs by default for all catalog members | Encourages usage of proper IDs. <br/>Can remove autoID conversion.<br/>Prevents changing IDs with `$magda-map-config-id` root group ID. | Makes catalog JSON less readable.<br/>Requires v7 autoID `shareKeys` | 
+
+Option 1: yes + yes for default
+
+Option 2: yes + give warning to users to not change IDs
+
+## Consequences
```
Comment (KeyboardSounds) [inline architecture/0005-root-group-ids-and-shareKeys.md:293]: Once you've filled out this section, I'll merge this 👍

**Review by KeyboardSounds — APPROVED** [review]

**Pull request merged**